handlers/deleteitem/deleteItem.go

```diff
@@ -0,0 +1,60 @@
+package deleteitem
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/mount-joy/thelist-lambda/db"
+	"github.com/mount-joy/thelist-lambda/handlers/iface"
+)
+
+type deleteItem struct {
+	db db.DB
+}
+
+// New returns an instance of deleteItem satisfying the RouteHandler interface
+func New() iface.RouteHandler {
+	return &deleteItem{
+		db: db.DynamoDB(),
+	}
+}
+
+// Match returns true if this RouteHandler should handle this request
+func (g *deleteItem) Match(request events.APIGatewayV2HTTPRequest) bool {
+	var re = regexp.MustCompile(`^/lists/([\w-]+)/items/([\w-]+)/?$`)
+	return request.RequestContext.HTTP.Method == "DELETE" && re.MatchString(request.RequestContext.HTTP.Path)
+}
+
+// Handle handles this request and returns the response and status code
+func (g *deleteItem) Handle(request events.APIGatewayV2HTTPRequest) (interface{}, int) {
+	listID, itemID, err := getIDs(request.RequestContext.HTTP.Path)
+	if err != nil {
+		log.Printf("Error: %s", err.Error())
+		return nil, http.StatusBadRequest
+	}
+
+	err = g.db.DeleteItem(listID, itemID)
+	if err != nil {
+		if aerr, ok := err.(db.Error); ok {
+			if aerr.ErrorType == db.ErrorNotFound {
+				return nil, http.StatusNotFound
+			}
+		}
+		log.Printf("Error: %s", err.Error())
+		return nil, http.StatusInternalServerError
+	}
+
+	return nil, http.StatusOK
+}
+
+func getIDs(path string) (*string, *string, error) {
```

Comment:
Wonder if a function with the signature: `func getElementFromPath(elementNumber int, path string) (string, error)`
Could be used for all routes, but probably appropriate for a different PR

Reply:
Yes I agree, let's fix it outside of this PR though